src/Dex/Foreign/Context.hs

```diff
@@ -33,7 +33,7 @@ import PPrint
 
 import Dex.Foreign.Util
 
-data Context = Context EvalConfig TopEnv
+data Context = Context EvalConfig Bindings
```

Comment:
Hmm I wonder if we should still keep `TopEnv` in here. `Context` is a bit like a top-level scope that you can interact with, and it should remember which modules were imported in the past.

Reply:
Good call. That worked out quite nicely.